src/components/popover/use-popover.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { useEventListener } from '@blockstack/ui';
+import { useControlledHover } from '@common/hooks/use-controlled-hover';
+import { useFocus } from 'use-events';
+import { UsePopoverProps } from '@components/popover/types';
+
+export const usePopover = ({ length, triggerRef }: UsePopoverProps) => {
+  const timeoutRef = React.useRef<number | undefined>(undefined);
+  const removeFocusRef = React.useRef<number | undefined>(undefined);
```

Comment:
Any reason not to use `null` here?  This tends to be my preference, as it's clear, then, something's explictly been set as `null`, rather than just having not been defined and implictly being `undefined`.

Reply:
yeah should probs be null, i tend to stick with undefined due to ts generally liking it better, and i must have just opted for that here instead of null. i can go through and ensure all the refs are default to null and fall back to it as the option